debezium-core/src/main/java/io/debezium/transforms/UnwrapFromEnvelope.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.transforms.ExtractField;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+
+/**
+ * Debezium generates CDC (<code>Envelope</code>) records that are struct of values containing values
+ * <code>before</code> and <code>after change</code>. Sink connectors usually are not able to work
+ * with a complex structure so a user use this SMT to extract <code>after</code> value and send it down
+ * unwrapped in <code>Envelope</code>.
+ * The functionality is similar to <code>ExtractField</code> SMT but has a special semantics for handling
+ * delete events.
+ *
+ * When delete event is emitted by database then Debezium emits two messages - delete message and tombstone
+ * message that serves as a signal to Kafka compaction process.
+ *
+ * The SMT by default drops tombstone message created by Debezium and converts delete message into a tombstone
+ * message that can be dropped too if required.
+ *
+ * @param <R> the subtype of {@link ConnectRecord} on which this transformation will operate
+ * @author Jiri Pechanec
+ */
+public class UnwrapFromEnvelope<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final String ENVELOPE_SCHEMA_NAME_SUFFIX = ".Envelope";
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final Field DROP_TOMBSTONES = Field.create("drop.tombstones")
+            .withDisplayName("Drop tombstones")
+            .withType(ConfigDef.Type.BOOLEAN)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withDefault(true)
+            .withDescription("Debezium by default generates a tombstone record to enable Kafka compaction after "
+                    + "a delete record was generated. This record is usually filtered out to avoid duplicates "
+                    + "as a delete record is converted to a tombstone record too");
+
+    private static final Field DROP_DELETES = Field.create("drop.deletes")
+            .withDisplayName("Drop outgoing tombstones")
+            .withType(ConfigDef.Type.BOOLEAN)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .withDefault(true)
```

Comment:
I'm actually thinking now that this config field should default to `false` so that deleted are passed through by default (and tombstones are dropped by default). It's not an obvious choice, because various sink connectors have differing capabilities - so I could be convinced the current default is preferred.

Reply:
So with the current default settings, does it mean DELETEs are not propagated? That seems not very desirable, does it?